### ISSUE TITLE
Fit Base Estimators with some data (Train Split #1) but fit Meta Estimator with the predictions of different data (Train Split #2)

### ISSUE THREAD

**Issue by ivsanro1** [rasbt/mlxtend#498]
I think there is a problem that if any of your Base Estimators overfits your training data, if the Meta Estimator is trained with the same data split, it will also overfit, because it will always select the accurate prediction for training that will come from the overfitted model.

One solution is to try to avoid models that overfit, but the best solution (because it generalizes more) is to train the Meta Estimator with the predicted probas of the trained Base Estimators but in another holded out training dataset (call it Training Dataset num 2), so it will be fitted with the truly best results of the Base Estimators, because if one Base Estimator is overfitted but does not perform well in validation, the Meta Estimator will not select it (or will give the overfitted Base Estimator a low weight).

tl;dr: title
How can I achieve this witl mlxtend's StackingClassifier?

Thanks in advance.

**Comment by rasbt** [comment]
> I think there is a problem that if any of your Base Estimators overfits your training data, if the Meta Estimator is trained with the same data split, it will also overfit, because it will always select the accurate prediction for training that will come from the overfitted model.

I agree! 

> How can I achieve this witl mlxtend's StackingClassifier?

I think you want to take a look at the StackingCVClassifier (not StackingClassifier) :)
